app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/service/StoreCharactersServiceImpl.kt

```diff
@@ -0,0 +1,17 @@
+package com.puzzlebench.clean_marvel_kotlin.data.service
+
+import com.puzzlebench.clean_marvel_kotlin.data.mapper.CharacterMapperRealm
+import com.puzzlebench.clean_marvel_kotlin.domain.contracts.CharacterRepository
+import com.puzzlebench.clean_marvel_kotlin.domain.model.Character
+import io.realm.Realm
+
+class StoreCharactersServiceImpl(private val mapper: CharacterMapperRealm = CharacterMapperRealm()) : CharacterRepository {
+
+    override fun saveCharactersDB(characters: List<Character>) {
+        val realm = Realm.getDefaultInstance()
+        realm.executeTransaction {
+            realm.deleteAll()
```

Comment:
it is necessary the deleteAll here? instead cannot be an update method?

Reply:
you can use the realm method **insertOrUpdate**

Reply:
Okey, thanks Mau!